pulsar-client/src/main/java/org/apache/pulsar/client/impl/ConsumerImpl.java

```diff
@@ -1248,6 +1254,77 @@ public void seek(MessageId messageId) throws PulsarClientException {
         return seekFuture;
     }
 
+    public boolean hasMessageAvailable() throws PulsarClientException {
+        try {
+            return hasMessageAvailableAsync().get();
+        } catch (ExecutionException | InterruptedException e) {
+            throw new PulsarClientException(e);
+        }
+    }
+
+    public CompletableFuture<Boolean> hasMessageAvailableAsync() {
+        final CompletableFuture<Boolean> booleanFuture = new CompletableFuture<>();
+
+        if (lastMessageIdInBroker.compareTo(lastDequeuedMessage) > 0 &&
```

Comment:
We should do this check also in the sync `hasMessageAvailable()` case, so that we can avoid creating a `CompletableFuture` each time we check, if the `lastMessageId` is already cached.

Reply:
Thanks. will add it.